### DISCUSSION TITLE
How can I conditionally mount/umount directories only for x86 / Pi

### DISCUSSION THREAD

**Discussion by intelliant01** [ltsp/ltsp#578] [Q&A] [answered]
Am trying to mount a directory only on x86 clients and not on Pi. For the same I tried these but without any success -

In ltsp.conf :
```
[clients]
FSTAB_01="server:/usr/local/share/mspython3 /usr/local/share/mspython3 nfs defaults,nolock,rsize=32768,wsize=32768 0 0"

[b8:27:eb:*|dc:a6:32:*|e4:5f:01:*]
POST_INIT_UMOUNT="umount -l /usr/local/share/mspython3 2>/dev/null || true"
```
Am not sure if `POST_INIT_*` may be used as I have done. Not sure if `[b8:27:eb:*|dc:a6:32:*|e4:5f:01:*]` section/case is indeed working. https://github.com/ltsp/ltsp/discussions/163#discussioncomment-101902 fuels my doubt further. :)

Even added unmount to the `/etc/ltsp/etc/rc.local` and `/usr/share/ltsp/client/99-umount.sh` and executed `ltsp initrd` to try and [serve the files via ltsp.img](https://ltsp.org/man/ltsp-initrd/) : 
```
umount -l /usr/local/share/mspython3 2>/dev/null || true
```

But it doesn't unmount.

`/etc/rc.local` doesn't get served dynamically so I put it directly in the Pi image and recreated it but yet it doesn't unmount.

This sequence only mounts the directory in all clients but doesn't unmount it in Pi. **The mounting of one directory in Pi, as per systemd-analyze seems to take 6+ seconds which is alarming as well.** However, on x86 client it mounts within 800 ms.
 
Need a way to only mount the directory in x86 systems or a way I can unmount them in Pi systems. 
Also why does it take so long to mount in Pi?

**Comment by alkisg** [comment, marked as answer]
What you're doing and fails is:

1. FSTAB_01 puts an entry in /etc/fstab
2. One second later, POST_INIT_UMOUNT tries to unmount it, but it's not mounted yet
3. Ten seconds later, systemd parses /etc/fstab and mounts it

One way around it is:

```shell
[clients]
FSTAB_01="server:/usr/local/share/mspython3 /usr/local/share/mspython3 nfs defaults,nolock,rsize=32768,wsize=32768 0 0"

[b8:27:eb:*|dc:a6:32:*|e4:5f:01:*]
POST_INIT_REMOVE_MOUNT="sed '/mspython3/d' -i /etc/fstab"
```

But after removing it, try to measure the actual difference that it made to your boot process. I bet it's under one second. The "6+ seconds" are time elapsed, which is parallelized; it's not CPU time.

**Reply by intelliant01** [reply]
Thanks @alkisg for the elegant solution.

Yes, difference is just 1 second now with and wihout the mount on the RPi.

But the difference in mounting on a x86 client and RPi client is more than 5 seconds as per systemd-analyze which intrigues me.

**Comment by alkisg** [comment]
Eh, a recent i3 is 50 times faster than a recent rpi... :)

**Reply by intelliant01** [reply]
Possibly the Celeron n3060 client I am pitching the RPi against is a tad too quick. :)
